File: src/lightnap-ng/src/app/core/backend-api/dtos/identity/request/verify-code-request-dto.ts
/**
 * Represents a request to verify a two-factor code.
 */
export interface VerifyCodeRequestDto {
    /**
     * The email address or user name associated with the verification request.
     */
    login: string;

    /**
     * The verification code to be validated.
     */
    code: string;

    /**
     * Indicates whether the user should be remembered on the device.
     */
    rememberMe: boolean;

    /**
     * Details about the device from which the request is made.
     */
    deviceDetails: string;
}

File: src/LightNap.Core/Identity/Dto/Request/VerifyCodeRequestDto.cs
using LightNap.Core.Configuration;
using System.ComponentModel.DataAnnotations;

namespace LightNap.Core.Identity.Dto.Request
{
    /// <summary>
    /// Data transfer object for verifying a 2FA code.
    /// </summary>
    public class VerifyCodeRequestDto
    {
        /// <summary>
        /// The email address or user name of the user.
        /// </summary>
        [Required]
        [StringLength(Constants.Dto.MaxLoginLength)]
        public required string Login { get; set; }

        /// <summary>
        /// The verification code.
        /// </summary>
        [Required]
        [StringLength(Constants.Dto.MaxVerificationCodeLength)]
        public required string Code { get; set; }

        /// <summary>
        /// True to remember the user on this device.
        /// </summary>
        public bool RememberMe { get; set; }

        /// <summary>
        /// The details of the device.
        /// </summary>
        [Required]
        [StringLength(Constants.Dto.MaxDeviceDetailsLength)]
        public required string DeviceDetails { get; set; }
    }
}